### PR TITLE
AF: vol-14x + eval-every-3 (max vol emphasis)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
Vol-weight=14x is the highest reasonable vol emphasis — beyond 15x, gilbert's prior run diverged. Combined with eval-every-3 for more epochs. Goal: push vol_mse below 0.002039 even if surface doesn't improve.

## Instructions
```bash
cd target/icml2026 && python train.py --dataset airfrans --optimizer lion --lr 1.25e-4 \
  --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 --enable-fourier \
  --epochs 999 --ema-decay 0.999 --use-lookahead --compile-model \
  --vol-loss-weight 14 --eval-interval 3 --save-checkpoint \
  --wandb-group af-eval-every-3-vol-sweep
```
**Target:** surface_mse < 0.000296 AND vol_mse < 0.002039

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT gohan:

## Results

### Best Metrics (at saved checkpoint, epoch 153)
| Metric | Baseline | Best | Epoch | vs Baseline |
|--------|----------|------|-------|-------------|
| surface_mse | 0.000296 | 0.001916 | 153 | 6.5x above target |
| vol_mse | 0.002039 | 0.006877 | 153 | 3.4x above target |

### Best Volume (not at checkpoint epoch)
- Best vol_mse across all evals: **0.00613** at epoch 189

### Key Observations
- **Neither target was met.** Vol-weight=14 is too aggressive — the model diverged catastrophically at epoch 191.
- **Divergence cascade:** Grad norms spiked 62 → 317 → 738 → 1117 → 18455 starting at epoch 190 (cosine restart), amplified by the 14x volume loss weighting. By epoch 195, surface_mse=0.110, vol_mse=0.223.
- **Volume was improving steadily before divergence:** vol_mse dropped from 0.55 → 0.006 over 189 epochs, setting new records almost every eval window.
- **Surface converged more slowly** than lower vol-weight runs — best surface was 0.00192 (6.5x above baseline), suggesting the 14x vol emphasis starves surface learning.
- **The 14x weight is near the instability boundary** — the advisor noted gilbert's prior run diverged at 15x. This run confirms 14x also diverges, just later (~epoch 190 vs earlier for 15x).

### Convergence Summary (pre-divergence)
| Epoch | surface_mse | vol_mse | Notes |
|-------|-------------|---------|-------|
| 1 | 0.4218 | 0.5525 | Start |
| 50 | 0.0095 | 0.0346 | End cycle 1 |
| 100 | 0.0046 | 0.0121 | End cycle 2 |
| 120 | 0.0025 | 0.0096 | |
| 138 | 0.0021 | 0.0077 | New best surface |
| 153 | 0.0019 | 0.0069 | **Best checkpoint** |
| 171 | 0.0020 | 0.0065 | |
| 189 | 0.0029 | 0.0061 | Best volume |
| 191 | — | — | **Divergence starts** |

### Experiment Details
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset airfrans --optimizer lion --lr 1.25e-4 \
  --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 --enable-fourier \
  --epochs 999 --ema-decay 0.999 --use-lookahead --compile-model \
  --volume-loss-weight 14 --eval-interval 3 --save-checkpoint \
  --wandb-group af-eval-every-3-vol-sweep \
  --agent gohan --wandb-name "gohan/af-vol14x-eval3"
```

- **W&B run:** h5es7gil
- **Peak GPU memory:** ~53.8 GB
- **Epochs completed:** 196+ (diverged at 191)

### What happened
The 14x vol-weight pushes volume learning aggressively and produces steady convergence through ~190 epochs. However, the extreme loss weighting amplifies gradient instabilities at cosine restarts, causing terminal divergence. The model never reached either baseline target before diverging.

### Suggested follow-ups
1. **Lower vol-weight (10-12x) with eval-every-3** may be the sweet spot — strong vol emphasis without hitting the instability ceiling.
2. **Gradient spike detection + skip** (skip-update-grad-norm) could prevent the cascading divergence that killed this run.
3. **Checkpoint recovery at epoch 153** with a lower vol-weight for fine-tuning could salvage the good representation learned.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3342 because ADVISOR: Closing — vol-14x diverged at ep191 (cosine restart gradient explosion, norm 62→18455). Surface 6.5x worse, vol 3.4x worse. Vol-13x (#3341) and vol-14x both unstable — the stable range is 10-12x. Vol-15x/20x also diverged (#3238). Gohan reassigned..
